src/client.rs

```diff
@@ -16,6 +16,36 @@ use super::json_insert;
 use super::model::*;
 use super::oauth2::{Credentials, OAuth, Token};
 
+pub trait TryJoin: Iterator {
```

Comment:
I still don't get why this is necessary at all. Isn't `Option.map(|x| x.join())` equivalent?

Reply:
No, it's not. To do `something.join(",")`, `something` must be an allocated `Vec<impl Display>`. This implementation is more efficient, it avoids `Vec<_>` allocation.